http/file_server_test.ts

```diff
@@ -498,3 +503,383 @@ Deno.test("file_server should show .. if it makes sense", async function (): Pro
     await killFileServer();
   }
 });
+
+Deno.test(
+  "file_server should download first byte of `hello.html` file",
+  async () => {
+    await startFileServer();
+    try {
+      const headers = {
+        "range": "bytes=0-0",
+      };
+      const res = await fetch(
+        "http://localhost:4507/testdata/test%20file.txt",
+        { headers },
+      );
+      const text = await res.text();
+      console.log(text);
+      assertEquals(text, "L");
+    } finally {
+      await killFileServer();
+    }
+  },
+);
+
+Deno.test(
+  "file_server sets `content-range` header for range request responses",
+  async () => {
+    await startFileServer();
+    try {
+      const headers = {
+        "range": "bytes=0-100",
+      };
+      const res = await fetch(
+        "http://localhost:4507/testdata/test%20file.txt",
+        { headers },
+      );
+      const contentLength = await getTestFileSize();
+      assertEquals(
+        res.headers.get("content-range"),
+        `bytes 0-100/${contentLength}`,
+      );
+
+      await res.text(); // Consuming the body so that the test doesn't leak resources
+    } finally {
+      await killFileServer();
+    }
+  },
+);
+
+const getTestFileSize = async () => {
+  const fileInfo = await getTestFileStat();
+  return fileInfo.size;
+};
+
+const getTestFileStat = async (): Promise<Deno.FileInfo> => {
+  const fsPath = join(testdataDir, "test file.txt");
+  const fileInfo = await Deno.stat(fsPath);
+
+  return fileInfo;
+};
+
+const getTestFileEtag = async () => {
+  const fileInfo = await getTestFileStat();
+
+  if (fileInfo.mtime instanceof Date) {
+    const lastModified = new Date(fileInfo.mtime);
+    const simpleEtag = await createEtagHash(
+      `${lastModified.toJSON()}${fileInfo.size}`,
+    );
+    return simpleEtag;
+  } else {
+    return "";
+  }
+};
+
+const createEtagHash = async (message: string) => {
+  // see: https://developer.mozilla.org/en-US/docs/Web/API/SubtleCrypto/digest
+  const hashType = "SHA-1"; // Faster, and this isn't a security senitive cryptographic use case
+  const msgUint8 = new TextEncoder().encode(message);
+  const hashBuffer = await crypto.subtle.digest(hashType, msgUint8);
+  const hashArray = Array.from(new Uint8Array(hashBuffer));
+  const hashHex = hashArray.map((b) => b.toString(16).padStart(2, "0")).join(
+    "",
+  );
+  return hashHex;
+};
```

Comment:
This code is duplicated... Seems better if you export it.

Also the `.join("",)` seems odd...

Reply:
I can move that out into a module. Is there guidance on where I should put that in the repo?

As for the `join`. Agreed `.join("",)` looks odd. But do you mean the trailing comma or joining the array into strings using `join`? The comma and line break are being added by `deno fmt`. I don't understand why it is doing that though. That method only accepts 1 argument.

Reply:
I pushed a commit to make `createEtagHash` clearer, and format better. It doesn't address moving it out into a module yet.

Reply:
@ry Any guidance on where I should put the module? Should I put it in something like `_createEtagHash.ts`?

Reply:
That would work. But I question whether the test needs the ability to create the etag header generally - how about just hardcoding some expected etag values into the test?

Reply:
My first tests did used hard coded values but it didn't work. I learned that the last modified time, and the system reported file size, aren't consistent across environments. Those are the two inputs, concatenated, for `createEtagHash`.